### PR TITLE
feat: search page (#1812)

### LINKED ISSUE #1812: UI (entity-search): Search Page
## Description

Display the result of the entity search as mockup attached.
The user must be able to search for entities, filter and sorting the results.

Integrate the logic for the searching [here](https://github.com/SwissDataScienceCenter/renku-ui/pull/1780) with UI for it #1808 and #1809 #1810

Screenshot 2022-04-27 at 14 55 33

Screenshot 2022-04-27 at 14 55 59

### PR DESCRIPTION
**Pull request by andre-code**
PR to integrate UI entity search and entity search Api.

Changes:
- Updated the My Projects and My Datasets links to redirect to the entity search page.
- Refactored quick navigation to be a functional component that uses the Redux toolkit dispatch hook
- Removal of auto-suggested projects in the search field
- Integrated author, type and visibility filters with entity search query
- Added filter display for small devices

SEARCH PAGE

SEARCH PAGE small devices

Closes #1812 
/deploy #persist

### CODE REVIEW

File: client/src/utils/components/searchResultsContent/SearchResultsContent.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  };
+
+  const phraseText = (<p>
+    We could not find any matching for phrase <span className="fst-italic rk-search-result-title">{phrase}</span>
+  </p>);
+
+  return (
+    <div className="mt-5 text-center">
+      <FontAwesomeIcon icon={faSadCry} size="3x" className="opacity-25" />{" "}
+      { phrase ? phraseText : " We could not find any matching." }
```
Comment (ciyer) [inline client/src/utils/components/searchResultsContent/SearchResultsContent.tsx:56]:
```suggestion
      { phrase ? phraseText : " We could not find any matches." }
```

**Review by ciyer — CHANGES_REQUESTED** [review]
This is a big step towards offering a much better search experience! 🎉 

In general, it looks good and works very well.

I have some suggestions for small changes to the code and wording, which I have mentioned in the appropriate place. And here are some things I noticed in the UI.

### No search results found

(I like the sad face icon!)
image

### Filter on mobile

The pop-up that goes to the left, right, and bottom edges of the screen looks strange. I think there should be some margin space on those sides.

image

### Author filter

I do not think this filter does anything.

image

File: client/src/utils/components/quicknav/QuickNav.container.js [hunk 2]
```diff
@@ -16,175 +16,90 @@
...
-    if (this.props.user.logged) {
-      const featured = this.projectsCoordinator.model.get("featured");
-      if (!featured.featured && !featured.fetching)
-        this.projectsCoordinator.getFeatured();
-    }
+const QuickNavContainerWithRouter = ({ user }) => {
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { phrase } = useKgSearchFormSelector((state) => state.kgSearchForm);
+  const [currentPhrase, setCurrentPhrase] = useState("");
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/utils/components/quicknav/QuickNav.container.js:53]: yes, it is necessary. The currentPhrase state is used to store the phrase that is updated each time the user types while KgSearchForm slice state is updated only when the text is ready to be queried (submit event), i.e. when the user presses enter or clicks the search button.

**Comment by andre-code** [comment]
Thanks @ciyer for your review and suggestions. 😉
You are right, there was a bug with the query so the author was not working, fixed in the last commit.
I have also included all the suggestions about styles and texts.

**Review by ciyer — CHANGES_REQUESTED** [review]
This is very close! I found a few more things in testing.

image

File: client/src/utils/components/sortingEntities/SortingEntities.tsx [hunk 3]
```diff
@@ -43,45 +46,45 @@ interface SortingItems {
...
 
   return (
     <>
-      <div className={styleType === "desk" ? "d-flex align-items-center" : ""}>
-        {styleType === "desk" ? <Label className="mx-2">Sort By</Label> : <h3 className="sorting-label">Sort By</h3>}
+      <div className={styleType === "desk" ?
+        "align-items-center d-none d-sm-none d-md-none d-lg-flex d-xl-flex d-xxl-flex"
+        : "sorting--mobile"}>
+        {styleType === "desk" ?
+          <Label className="mx-2 sorting-label--desk">Sort By</Label> :
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/utils/components/sortingEntities/SortingEntities.tsx]: sure, done

File: client/src/utils/components/entitySearchFilter/EntitySearchFilter.tsx [hunk 4]
```diff
@@ -28,38 +33,30 @@ import { VisibilityFilter } from "../visibilityFilter/VisibilityFilter";
...
+const FilterEntitySearch = ({ author, type, visibility, isLoggedUser }: FilterProps) => {
+  const dispatch = useDispatch();
+  const authorComponent = isLoggedUser ? (
+    <div><AuthorFilter
+      handler={(value: KgAuthor) => dispatch(setAuthor(value))}
+      value={author} /></div>
+  ) : null;
   return (
     <>
+      <h3 className="filter-label d-none d-sm-none d-md-none d-lg-block d-xl-block pb-3">Filter by</h3>
```
[the remark this answers is not in the document]
Comment (andre-code) [reply client/src/utils/components/entitySearchFilter/EntitySearchFilter.tsx]: sure, done

**Comment by andre-code** [comment]
> This is very close! I found a few more things in testing.
> 


Thanks @ciyer for your review, find the text hierarchy fix in the last commit.

**Review by ciyer — APPROVED** [review]
Looks great and works well on a variety of viewport sizes!

**Pull request merged**